harvard-lil/capstone-static
# Add about page

## What this does 
Adds an about page, plus many supporting elements. 

## Summary 
- Adds `about.html`, plus a web component template, `cap-about-page`
- Adds `cap-anchor-list` component, for the sidebar links shown on many interior pages
- Adds `cap-page-header`, for all color-blocked interior page headers
- Adds `cap-contributor-list` component, for the many past and present contributors to CAP
- Updates `cap-decorated-header` to support h1 or h2 headings, multiple sizes of heading, different decorators, 
- Refined styling for `cap-media-list`
- Renamed `cap-header` to `cap-nav` to avoid confusion with page header components
- Added quite a large number of root element styles, utility classes, one layout class, and unscoped component css styles  for adhoc styling on interior pages. If you're interested about the strategy behind this, you can read more in the Harry Roberts article [BEMIT: taking the BEM naming convention a step further](https://csswizardry.com/2015/08/bemit-taking-the-bem-naming-convention-a-step-further/).
  -  Utility classes are prefixed `u`
  -  Layout classes  are prefixed `l` 
  - Component styles are prefixed `c`
 - Refined all global responsive design styles, across components shared between the about page and the homepage and root/global css selectors
 - Added three new font-size css variables
 - Added supporting JavaScript files containing structured data for the about sidebar links as well as each contributor to CAP
 - Removed unused tooltip JavaScript
 
## Screenshots
Desktop view: 

![Screenshot 2023-12-20 at 4 05 45 PM](https://github.com/harvard-lil/capstone-static/assets/4039311/30b64440-4a2d-4bec-8f1b-9cc3dcc22925)

Smaller mobile view: 
![Screenshot 2023-12-20 at 4 05 58 PM](https://github.com/harvard-lil/capstone-static/assets/4039311/f83a83ee-e2d9-4115-930d-a40143284396)


## How to test
- Add a new remote for this fork 
- Checkout this branch
- Spin up your favorite webserver (the default one, used by the VSCode-suggested server plugin, will spin up the homepage at `https://localhost:5501/`
- Visit https://localhost:5501/about.html — note that the ".html" is crucial!
